### PR TITLE
Added databases/postgresql.md

### PR DESCRIPTION
**Pull request by syndbg**
Just something that I missed in the `sinatra-recipes`.

This is one of the few documents I have ready. Others are `data-mapper`, `activerecord` and `mysql`. But one by one.

### CODE REVIEW

File: databases/postgresql.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+[PostgreSQL](http://www.postgresql.org/) is an object-relational database management system (ORDBMS).
+
+The connection between PostgreSQL and Ruby happens via the [`pg`](https://rubygems.org/gems/pg) gem.
+
+## Installation
+
+[Install PostgreSQL](http://www.postgresql.org/download/)
+
+```
+gem install pg
```
Comment (kgrz) [inline databases/postgresql.md]: Does this need some description? something like "After the DB in installed on the machine, install the driver" etc.?

File: databases/postgresql.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+```
+
+**ActiveRecord or DataMapper**. You need only one, but we'll show both.
```
Comment (kgrz) [inline databases/postgresql.md]: Is it a good idea to split this into two separate recipes?

File: databases/postgresql.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+      t.timestamps null: false
+    end
+  end
+end
+```
+
+
+**CRUD operations in routes**
+
+```ruby
```
Comment (kgrz) [inline databases/postgresql.md]: We should probably add a `require 'models/*'` (or something similar) line here. Best to show the folder structure somewhere upfront.

File: databases/postgresql.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+end
+```
+
+**Create migrations**
+
+First, create a `Rakefile` or update your `Rakefile` to require `activerecord/rake`
+
+```
+require 'sinatra/activerecord/rake'
+...
```
Comment (kgrz) [inline databases/postgresql.md]: extraneous line?

**Comment by kgrz** [comment]
Thank you so much for this!

File: databases/postgresql.md (hunk 4 above)
Comment (syndbg) [inline databases/postgresql.md]: The require line is needed to expose the `sinatra-activerecord` gem rake functionality.
```
rake db:create
rake db:drop
rake db:migrate
..... and etc
```

File: databases/postgresql.md (hunk 3 above)
Comment (syndbg) [inline databases/postgresql.md]: Agreed! Will update it

File: databases/postgresql.md (hunk 2 above)
Comment (syndbg) [inline databases/postgresql.md]: Sounds like a good idea. The post is too long to follow and they can just choose their preferred ORM from http://recipes.sinatrarb.com/p/databases?#article . Will update it accordingly

File: databases/postgresql.md (hunk 1 above)
Comment (syndbg) [inline databases/postgresql.md]: It certainly does need a description. I threw the line there without giving the reader much reason for it. I will update it !

**Comment by syndbg** [comment]
Updated! Feel free to review it again when you have time :metal:

**Comment by syndbg** [comment]
Ping :1234:

File: databases/postgresql-activerecord.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+```
+require 'sinatra/activerecord/rake'
+...
+```
+
+Then create a migration as shown:
+
+```
+▶ rake db:create_migration NAME=create_articles
+db/migrate/20150516144225_create_articles.rb
```
Comment (kgrz) [inline databases/postgresql-activerecord.md]: This looks a bit ambiguous. It looks as if we're asking to type-in the `db/migrate/...` part too. Do you mind if I fix this?

File: databases/postgresql-activerecord.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+or by specifying in your sinatra app
+
+```ruby
+configure :development do
+  set :database, {adapter: 'postgres',  encoding: 'unicode', database: 'your_database_name', pool: 2, username: 'your_username', password: 'your_password'}
+end
+
+configure :production do
+  set :database, {adapter: 'postgres',  encoding: 'unicode', database: 'your_database_name', pool: 2, username: 'your_username', password: 'your_password'}
+end
```
Comment (kgrz) [inline databases/postgresql-activerecord.md]: Are you sure this is going to work? Doesn't AR's migration `rake` helper require the `config/database.yml` file to be present? If AR does recognize Sinatra's configuration helper, then there probably should be a specific way to load it into the `Rakefile`.

**Comment by kgrz** [comment]
@syndbg ⏰

**Comment by syndbg** [comment]
Alright, I really took my time. 

Just tested and it works just fine with `configure` block instead of `config/database.yml`. The tricky part was that it should be `postgresql` instead of `postgres` and in the `Rakefile` you should also require `./app`. 

Everything else is A-okay. :beetle:

**Comment by kgrz** [comment]
Hmm. I tried setting up an app with datamapper and followed the first part of setup (without the AR-style setup) and I'm getting an error specifying that the database is not setup, and I couldn't find any instructions. Checkout this repo: https://github.com/kgrz/datamapper

If I run `require_relative 'app.rb'`, I get an error saying the database is not setup. Some more notes:

```
Sinatra AR

1. Add comments about model directory path
2. in the get index route, `@articles = Articles.all` is called. But this is supposed
   to be `Article.all`
3. Requiring `sinatra/activerecord` is a must to ensure the rake tasks
   run properly


Datamapper

1. Missing dm-migrations gem install
2. Missing instruction to create the DB. `auto_migrate!` doesn’t work
   without an existing DB.
```

**Comment by syndbg** [comment]
@kgrz 

> Sinatra AR

> 1. Add comments about model directory path

Done.

> 2. in the get index route, `@articles = Articles.all` is called. But this is supposed to be `Article.all`

Fixed in both files.

> 3. Requiring `sinatra/activerecord` is a must to ensure the rake tasks run properly

Added.

> Datamapper

> 1. Missing dm-migrations gem install

Replaced the `dm-core` and `dm-migrations` with `data_mapper` which installs everything except database adapters, needed to use `DataMapper`.

> 2. Missing instruction to create the DB. `auto_migrate!` doesn’t work without an existing DB.

Yep, they were missing. Turns out you can't create automagically a DB with `DataMapper`.
Added instructions about that too.

**Comment by syndbg** [comment]
@kgrz No worries. 👍

**Pull request merged**